src/Components/ColorPills/ColorPills.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from 'react';
+import {
+    IColorPillsProps,
+    IColorPillsStyleProps,
+    IColorPillsStyles
+} from './ColorPills.types';
+import {
+    classNamesFunction,
+    IStackTokens,
+    Stack,
+    styled
+} from '@fluentui/react';
+import { useId } from '@fluentui/react-hooks';
+import { useExtendedTheme } from '../../Models/Hooks/useExtendedTheme';
+import { getColorPillsStyles } from './ColorPills.styles';
+
+const getClassNames = classNamesFunction<
+    IColorPillsStyleProps,
+    IColorPillsStyles
+>();
+
+const ColorPills: React.FC<IColorPillsProps> = (props) => {
+    const { visualColorings, width, styles } = props;
+    const stackTokens: IStackTokens = { childrenGap: 2 };
+    const alignment = width === 'wide' ? 'center' : 'start';
+    const pillId = useId('cb-visual-rule-pills');
+
+    // styles
+    const classNames = getClassNames(styles, {
+        theme: useExtendedTheme(),
+        width: props.width
+    });
+
+    const [trimmedVisualColorings, setTrimmedVisualColorings] = useState(
+        visualColorings && visualColorings.length > 3
```

Comment:
since you already updating it in useEffect down below with the changes of visualColortins, I don't think you should set a default value here - repetition of the code